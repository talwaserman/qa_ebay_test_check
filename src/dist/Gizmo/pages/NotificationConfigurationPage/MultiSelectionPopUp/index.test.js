"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _regenerator = _interopRequireDefault(require("@babel/runtime/regenerator"));

var _asyncToGenerator2 = _interopRequireDefault(require("@babel/runtime/helpers/asyncToGenerator"));

var _react = _interopRequireDefault(require("react"));

var _reactTestRenderer = _interopRequireDefault(require("react-test-renderer"));

var _sortBy = _interopRequireDefault(require("lodash/sortBy"));

var _get = _interopRequireDefault(require("lodash/get"));

var _cloneDeep = _interopRequireDefault(require("lodash/cloneDeep"));

var _services = require("../services");

var _mock = require("../mock");

var _index = _interopRequireDefault(require("./index"));

require("regenerator-runtime/runtime");

// - TODO we might need setupTest.js file - or do we have one ?
var multiSelectionPopUpData = {
  placeHolder1: 'Type',
  placeHolder2: 'Activity',
  placeHolder3: 'Assignees',
  selectionData: _mock.getNotificationTypeResponse
};

function getSortedUsersData() {
  return _getSortedUsersData.apply(this, arguments);
}

function _getSortedUsersData() {
  _getSortedUsersData = (0, _asyncToGenerator2.default)( /*#__PURE__*/_regenerator.default.mark(function _callee() {
    var usersDataResponse, sortedUsersDataResponse;
    return _regenerator.default.wrap(function _callee$(_context) {
      while (1) {
        switch (_context.prev = _context.next) {
          case 0:
            _context.next = 2;
            return (0, _services.getUsers)();

          case 2:
            usersDataResponse = _context.sent;
            sortedUsersDataResponse = (0, _cloneDeep.default)((0, _get.default)(usersDataResponse, 'users', null));
            sortedUsersDataResponse = (0, _sortBy.default)(sortedUsersDataResponse, function (value) {
              return value.display.toUpperCase();
            });
            return _context.abrupt("return", sortedUsersDataResponse);

          case 6:
          case "end":
            return _context.stop();
        }
      }
    }, _callee);
  }));
  return _getSortedUsersData.apply(this, arguments);
}

it('Gizmo shared - default template test for generating MultiSelectionPopUp for add new notification', function () {
  var tree = _reactTestRenderer.default.create( /*#__PURE__*/_react.default.createElement(_index.default, {
    multiSelectionPopUpData: multiSelectionPopUpData,
    dropDownKeyLabel1: 'ktype',
    dropDownKeyLabel2: 'kactivity',
    dropDownKeyLabel3: 'kassignees',
    componentTitle: 'Add New Notification',
    multiSelectionTitle: 'Assignees',
    multiSelectionData: getSortedUsersData
  })).toJSON();

  expect(tree).toMatchSnapshot();
});
it('Gizmo shared - default template test for generating MultiSelectionPopUp for editing existing notification', function () {
  var tree = _reactTestRenderer.default.create( /*#__PURE__*/_react.default.createElement(_index.default, {
    multiSelectionPopUpData: multiSelectionPopUpData,
    dropDownKeyLabel1: 'ktype',
    dropDownKeyLabel2: 'kactivity',
    dropDownKeyLabel3: 'kassignees',
    componentTitle: 'Edit Notification',
    multiSelectionTitle: 'Assignees',
    multiSelectionData: getSortedUsersData,
    existingData: _mock.notificationsListMock,
    isDisabledFirstSelection: true,
    isDisabledSecondSelection: true,
    editSelectedData: {
      type: _mock.notificationsListMock.notifications[0].type,
      activity: _mock.notificationsListMock.notifications[0].activity,
      users: _mock.notificationsListMock.notifications[0].users
    }
  })).toJSON();

  expect(tree).toMatchSnapshot();
});