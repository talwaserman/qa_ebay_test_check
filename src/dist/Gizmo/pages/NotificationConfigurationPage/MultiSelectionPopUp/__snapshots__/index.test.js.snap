// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gizmo shared - default template test for generating MultiSelectionPopUp for add new notification 1`] = `
<div
  className="multi-selection-popup-wrapper"
>
  <div
    className="add-activity-label"
  >
    Type
  </div>
  <div
    className="notification-type-select"
  >
    <div
      className="ant-select ant-select-enabled"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseOut={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
    >
      <div
        aria-autocomplete="list"
        aria-controls="test-uuid"
        aria-expanded={false}
        aria-haspopup="true"
        className="ant-select-selection
            ant-select-selection--single"
        onKeyDown={[Function]}
        role="combobox"
        tabIndex={0}
      >
        <div
          className="ant-select-selection__rendered"
        >
          <div
            className="ant-select-selection__placeholder"
            onClick={[Function]}
            onMouseDown={[Function]}
            style={
              Object {
                "WebkitUserSelect": "none",
                "display": "block",
                "userSelect": "none",
              }
            }
            unselectable="on"
          >
            Type
          </div>
        </div>
        <span
          className="ant-select-arrow"
          onClick={[Function]}
          style={
            Object {
              "WebkitUserSelect": "none",
              "userSelect": "none",
            }
          }
          unselectable="on"
        >
          <i
            aria-label="icon: down"
            className="anticon anticon-down ant-select-arrow-icon"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="down"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
              />
            </svg>
          </i>
        </span>
      </div>
    </div>
  </div>
  <div
    className="add-activity-label"
  >
    Activity
  </div>
  <div
    className="activity-type-select"
  >
    <div
      className="ant-select ant-select-enabled"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseOut={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
    >
      <div
        aria-autocomplete="list"
        aria-controls="test-uuid"
        aria-expanded={false}
        aria-haspopup="true"
        className="ant-select-selection
            ant-select-selection--single"
        onKeyDown={[Function]}
        role="combobox"
        tabIndex={0}
      >
        <div
          className="ant-select-selection__rendered"
        >
          <div
            className="ant-select-selection__placeholder"
            onClick={[Function]}
            onMouseDown={[Function]}
            style={
              Object {
                "WebkitUserSelect": "none",
                "display": "block",
                "userSelect": "none",
              }
            }
            unselectable="on"
          >
            Activity
          </div>
        </div>
        <span
          className="ant-select-arrow"
          onClick={[Function]}
          style={
            Object {
              "WebkitUserSelect": "none",
              "userSelect": "none",
            }
          }
          unselectable="on"
        >
          <i
            aria-label="icon: down"
            className="anticon anticon-down ant-select-arrow-icon"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="down"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
              />
            </svg>
          </i>
        </span>
      </div>
    </div>
  </div>
  <div
    className="add-assigners-label"
  >
    Assignees
  </div>
  <div
    className="users-multi-select"
  >
    <div
      className="fast-multi-select-spiner-wrapper"
    >
      <div
        className="ant-spin ant-spin-spinning"
      >
        <i
          aria-label="icon: loading"
          className="anticon anticon-loading ant-spin-dot"
          style={
            Object {
              "fontSize": 24,
            }
          }
        >
          <svg
            aria-hidden="true"
            className="anticon-spin"
            data-icon="loading"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="0 0 1024 1024"
            width="1em"
          >
            <path
              d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 0 0-94.3-139.9 437.71 437.71 0 0 0-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
            />
          </svg>
        </i>
      </div>
      Loading ...
    </div>
  </div>
</div>
`;

exports[`Gizmo shared - default template test for generating MultiSelectionPopUp for editing existing notification 1`] = `
<div
  className="multi-selection-popup-wrapper"
>
  <div
    className="add-activity-label"
  >
    Type
  </div>
  <div
    className="notification-type-select"
  >
    <div
      className="ant-select ant-select-disabled"
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseOut={[Function]}
      onMouseUp={[Function]}
    >
      <div
        aria-autocomplete="list"
        aria-controls="test-uuid"
        aria-expanded={false}
        aria-haspopup="true"
        className="ant-select-selection
            ant-select-selection--single"
        onKeyDown={[Function]}
        role="combobox"
        tabIndex={-1}
      >
        <div
          className="ant-select-selection__rendered"
        >
          <div
            className="ant-select-selection__placeholder"
            onClick={[Function]}
            onMouseDown={[Function]}
            style={
              Object {
                "WebkitUserSelect": "none",
                "display": "none",
                "userSelect": "none",
              }
            }
            unselectable="on"
          >
            Type
          </div>
          <div
            className="ant-select-selection-selected-value"
            style={
              Object {
                "display": "block",
                "opacity": 1,
              }
            }
            title="externalTask"
          >
            externalTask
          </div>
        </div>
        <span
          className="ant-select-arrow"
          onClick={[Function]}
          style={
            Object {
              "WebkitUserSelect": "none",
              "userSelect": "none",
            }
          }
          unselectable="on"
        >
          <i
            aria-label="icon: down"
            className="anticon anticon-down ant-select-arrow-icon"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="down"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
              />
            </svg>
          </i>
        </span>
      </div>
    </div>
  </div>
  <div
    className="add-activity-label"
  >
    Activity
  </div>
  <div
    className="activity-type-select"
  >
    <div
      className="ant-select ant-select-disabled"
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseOut={[Function]}
      onMouseUp={[Function]}
    >
      <div
        aria-autocomplete="list"
        aria-controls="test-uuid"
        aria-expanded={false}
        aria-haspopup="true"
        className="ant-select-selection
            ant-select-selection--single"
        onKeyDown={[Function]}
        role="combobox"
        tabIndex={-1}
      >
        <div
          className="ant-select-selection__rendered"
        >
          <div
            className="ant-select-selection__placeholder"
            onClick={[Function]}
            onMouseDown={[Function]}
            style={
              Object {
                "WebkitUserSelect": "none",
                "display": "none",
                "userSelect": "none",
              }
            }
            unselectable="on"
          >
            Activity
          </div>
          <div
            className="ant-select-selection-selected-value"
            style={
              Object {
                "display": "block",
                "opacity": 1,
              }
            }
            title="activity ended"
          >
            activity ended
          </div>
        </div>
        <span
          className="ant-select-arrow"
          onClick={[Function]}
          style={
            Object {
              "WebkitUserSelect": "none",
              "userSelect": "none",
            }
          }
          unselectable="on"
        >
          <i
            aria-label="icon: down"
            className="anticon anticon-down ant-select-arrow-icon"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="down"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
              />
            </svg>
          </i>
        </span>
      </div>
    </div>
  </div>
  <div
    className="add-assigners-label"
  >
    Assignees
  </div>
  <div
    className="users-multi-select"
  >
    <div
      className="fast-multi-select-spiner-wrapper"
    >
      <div
        className="ant-spin ant-spin-spinning"
      >
        <i
          aria-label="icon: loading"
          className="anticon anticon-loading ant-spin-dot"
          style={
            Object {
              "fontSize": 24,
            }
          }
        >
          <svg
            aria-hidden="true"
            className="anticon-spin"
            data-icon="loading"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="0 0 1024 1024"
            width="1em"
          >
            <path
              d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 0 0-94.3-139.9 437.71 437.71 0 0 0-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
            />
          </svg>
        </i>
      </div>
      Loading ...
    </div>
  </div>
</div>
`;
